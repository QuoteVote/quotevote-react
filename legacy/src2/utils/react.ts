import React, { 
  useState, 
  useEffect, 
  useRef, 
  ReactNode, 
  ChangeEvent, 
  KeyboardEvent,
  FormEvent,
  MouseEvent,
  Dispatch,
  SetStateAction
} from 'react';

export type { 
  ReactNode,
  ChangeEvent,
  KeyboardEvent,
  FormEvent,
  MouseEvent,
  Dispatch,
  SetStateAction
};

export { 
  React as default,
  useState,
  useEffect,
  useRef
};

// Add type declarations for common event handlers
export type InputChangeEvent = ChangeEvent<HTMLInputElement>;
export type TextAreaChangeEvent = ChangeEvent<HTMLTextAreaElement>;
export type ButtonClickEvent = MouseEvent<HTMLButtonElement>;
export type SetState<T> = Dispatch<SetStateAction<T>>;

// Add type declarations for common component props
export interface BaseProps {
  className?: string;
  style?: React.CSSProperties;
  children?: ReactNode;
}

export interface InputProps extends BaseProps {
  value: string;
  onChange: (value: string) => void;
  placeholder?: string;
  disabled?: boolean;
} 