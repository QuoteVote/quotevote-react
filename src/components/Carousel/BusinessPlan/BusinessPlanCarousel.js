import React from 'react'
import { Typography } from '@material-ui/core'
import PropTypes from 'prop-types'
import GridItem from '../../../mui-pro/Grid/GridItem'
import GridContainer from '../../../mui-pro/Grid/GridContainer'
import businessPlanImg from '../../../assets/img/RequestAccess/Illustration.png'
import businessPlanImg2 from '../../../assets/img/Chatbox.png'
import businessPlanImg3 from '../../../assets/img/CommentBox.png'
import GetAccessButton from '../../CustomButtons/GetAccessButton'
import Carousel from '../Carousel'
import DoubleArrowIconButton from '../../CustomButtons/DoubleArrowIconButton'

BusinessCarouselFirstContent.propTypes = {
  classes: PropTypes.object,
  handleNext: PropTypes.func,
}

function BusinessCarouselFirstContent({ classes, handleNext }) {
  const { opinionsText, bottomText, greenText } = classes
  return (
    <GridContainer justify="center" style={{ marginRight: 24 }}>
      <GridItem xs={12} sm={5}>
        <GridContainer justify="center">
          <img
            alt={businessPlanImg}
            height={500}
            src={`${businessPlanImg}`}
            style={{
              width: '435.43px',
              height: '300.51px',
              objectFit: 'contain',
            }}
          />
        </GridContainer>
      </GridItem>
      <GridItem xs={12} sm={4}>
        <GridContainer justify="left">
          <Typography>
            <div className={opinionsText}>
              <b>Join 345 companies in creating a workspace in which everyone has an equal voice.</b>
              {' '}
              Poll to see what your employees value the most, converse to make the next big decision, and more.
              <br />
              <br />
              voxPOP is yours to shape.
              <br />
              <br />
            </div>
          </Typography>
          <Typography className={bottomText}>
            <GetAccessButton />
            {' '}
            What else do
            <span className={greenText}> we have </span>
            for you?
            <DoubleArrowIconButton onClick={handleNext} />
          </Typography>
        </GridContainer>
      </GridItem>
    </GridContainer>
  )
}

BusinessCarouselSecondContent.propTypes = {
  classes: PropTypes.object,
  handleNext: PropTypes.func,
}
function BusinessCarouselSecondContent({ classes, handleNext }) {
  const { opinionsText, bottomText, greenText } = classes
  return (
    <GridContainer justify="center" style={{ marginRight: 24 }}>
      <GridItem xs={12} sm={5}>
        <GridContainer justify="center">
          <img
            alt={businessPlanImg2}
            height={500}
            src={`${businessPlanImg2}`}
            style={{
              width: '435.43px',
              height: '300.51px',
              objectFit: 'contain',
            }}
          />
        </GridContainer>
      </GridItem>
      <GridItem xs={12} sm={4}>
        <GridContainer justify="left">
          <Typography>
            <div className={opinionsText}>
              <b>Talk to your team,</b>
              {' '}
              poll the entire company, assess their feedback, plan the next big company initiative that will knock it out of the park.
              <br />
              <br />
            </div>
          </Typography>
          <Typography className={bottomText}>
            <GetAccessButton />
            {'  '}
            Stay
            {' '}
            <span className={greenText}> in Sync</span>
            <DoubleArrowIconButton onClick={handleNext} />
          </Typography>
        </GridContainer>
      </GridItem>
    </GridContainer>
  )
}

function BusinessCarouselThirdContent(classes) {
  return (
    <GridContainer justify="center" style={{ marginRight: 24 }}>
      <GridItem xs={12} sm={5}>
        <GridContainer justify="center">
          <img
            alt={businessPlanImg3}
            height={500}
            src={`${businessPlanImg3}`}
            style={{
              width: '435.43px',
              height: '300.51px',
              objectFit: 'contain',
            }}
          />
        </GridContainer>
      </GridItem>
      <GridItem xs={12} sm={4}>
        <GridContainer justify="left">
          <Typography>
            <div className={classes.opinionsText}>
              <b>Promote democracy and transparency</b>
              {' '}
              within your team. VoxPop makes it so the whole team can see progress in real time.
              <br />
              <br />
            </div>
          </Typography>
          <GetAccessButton />
        </GridContainer>
      </GridItem>
    </GridContainer>
  )
}

function BusinessPlanCarousel(props) {
  const [activeStepProp, setActiveStepProp] = React.useState(0)
  const handleNext = () => {
    setActiveStepProp((prevActiveStep) => prevActiveStep + 1)
  }

  return (
    <Carousel
      autoplay={false}
      navButtonsAlwaysVisible={false}
      activeStepProp={activeStepProp}
      setActiveStepProp={setActiveStepProp}
    >
      <BusinessCarouselFirstContent {...props} handleNext={handleNext} />
      <BusinessCarouselSecondContent {...props} handleNext={handleNext} />
      <BusinessCarouselThirdContent {...props} handleNext={handleNext} />
    </Carousel>
  )
}

BusinessPlanCarousel.propTypes = {
  classes: PropTypes.object,
}

export default BusinessPlanCarousel
