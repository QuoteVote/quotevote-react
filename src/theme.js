import orange from '@material-ui/core/colors/orange'
// import blue from '@material-ui/core/colors/blue'
// import grey from '@material-ui/core/colors/grey'
import pink from '@material-ui/core/colors/pink'
import red from '@material-ui/core/colors/red'
import green from '@material-ui/core/colors/green'
// import yellow from '@material-ui/core/colors/yellow'
import cyan from '@material-ui/core/colors/cyan'
import teal from '@material-ui/core/colors/teal'

const theme = {
  activityCards: {
    quoted: {
      color: cyan[500],
    },
    commented: {
      color: orange[500],
    },
    upvote: {
      color: green[500],
    },
    downvote: {
      color: red[500],
    },
    submitted: {
      color: '#000000',
    },
    hearted: {
      color: pink,
    },
  },
  subHeader: {
    activeIcon: {
      color: teal.A400,
    },
    default: {
      color: 'black',
    },
    followButton: {
      backgroundColor: '#00CF6E',
      color: 'white',
    },
  },
}

export default theme
