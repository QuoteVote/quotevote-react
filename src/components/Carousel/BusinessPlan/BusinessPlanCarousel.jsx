import { useState } from 'react'
import { Typography } from '@material-ui/core'
import PropTypes from 'prop-types'
import Grid from '@material-ui/core/Grid'
import Carousel from 'react-material-ui-carousel'
import withWidth from '@material-ui/core/withWidth'
import { isMobile } from 'react-device-detect'
import RequestInviteCarouselButton from '../RequestInviteCarouselButton'
import { MOBILE_IMAGE_WIDTH } from '../../../views/LandingPage/LandingPage'

function BusinessCarouselFirstContent(props) {
  const { width, classes } = props
  const browserWidth = width === 'xs' ? '400px' : '80%'
  const imageWidth = isMobile ? MOBILE_IMAGE_WIDTH : browserWidth
  const padding = width === 'xs' ? 20 : 60

  return (
    <Grid
      container
      direction="row"
      justify="center"
      alignItems="center"
      style={{ paddingLeft: padding, paddingRight: padding }}
    >
      <Grid item sm={12} md={8} lg={7}>
        <img
          alt="Business"
          src="/assets/BusinessContent1.svg"
          style={{
            width: imageWidth,
            height: isMobile ? 'auto' : '350.51px',
            objectFit: 'contain',
          }}
        />
      </Grid>
      <Grid item sm={12} md={4} lg={5}>
        <Typography>
          <div className={classes.opinionsText}>
            <p>
              <b>Talk to your team</b>
              {' '}
              , poll the entire company, assess their feedback,
              plan the next big company initiative that will knock it
              out of the park.
            </p>
            <br />
          </div>
        </Typography>
      </Grid>
      <RequestInviteCarouselButton classes={classes} />
    </Grid>
  )
}

BusinessCarouselFirstContent.propTypes = {
  width: PropTypes.string,
  classes: PropTypes.object,
}

function BusinessCarouselSecondContent(props) {
  const { width, classes } = props
  const browserWidth = width === 'xs' ? '400px' : '80%'
  const imageWidth = isMobile ? 200 : browserWidth
  const padding = width === 'xs' ? 20 : 60
  return (
    <Grid
      container
      direction="row"
      justify="center"
      alignItems="center"
      style={{ paddingLeft: padding, paddingRight: padding }}
    >
      <Grid item sm={12} md={8} lg={7}>
        <img
          alt="Business 2"
          src="/assets/BusinessContent2.svg"
          style={{
            width: imageWidth,
            height: isMobile ? 'auto' : '350.51px',
            objectFit: 'contain',
          }}
        />
      </Grid>
      <Grid item sm={12} md={4} lg={5}>
        <Typography>
          <div className={classes.opinionsText}>
            <p style={{ marginTop: width === 'xs' ? 0 : 20 }}>
              <b>Promote democracy and transparency</b>
              {' '}
              within your team. Quote makes it so the whole team
              can see progress in real time.
            </p>
            <br />
          </div>
        </Typography>
      </Grid>
      <Grid
        item
        container
        justify="center"
        alignItems="center"
        sm={12}
      >
        <RequestInviteCarouselButton classes={classes} />
      </Grid>
    </Grid>
  )
}

BusinessCarouselSecondContent.propTypes = {
  width: PropTypes.string,
  classes: PropTypes.object,
}

function BusinessCarouselThirdContent(props) {
  const { width, classes } = props
  const browserWidth = width === 'xs' ? '400px' : '70%'
  const imageWidth = isMobile ? 200 : browserWidth
  const padding = width === 'xs' ? 20 : 60
  return (
    <Grid
      container
      direction="row"
      justify="center"
      alignItems="center"
      style={{ paddingLeft: padding, paddingRight: padding }}
    >
      <Grid item sm={12} md={8} lg={7}>
        <img
          alt="Business 2"
          src="/assets/BusinessContent3.png"
          style={{
            width: imageWidth,
            height: isMobile ? 'auto' : '350.51px',
            objectFit: 'contain',
          }}
        />
      </Grid>
      <Grid item sm={12} md={4} lg={5}>
        <Typography>
          <div className={classes.opinionsText}>
            <p style={{ marginTop: width === 'xs' ? 0 : 20 }}>
              <b>
                Join 345 companies in creating a workspace in
                which everyone has a voice.
              </b>
              {' '}
              Poll to see what your employees value the most,
              converse to make the next big decision, and more.
            </p>
            <br />
            <p>
              <span className={classes.greenText}>Quote</span>
              {' '}
              is yours to shape.
            </p>
          </div>
        </Typography>
      </Grid>
      <Grid
        item
        container
        justify="center"
        alignItems="center"
        sm={12}
      >
        <RequestInviteCarouselButton classes={classes} />
      </Grid>
    </Grid>
  )
}

BusinessCarouselThirdContent.propTypes = {
  width: PropTypes.string,
  classes: PropTypes.object,
}

function BusinessPlanCarousel(props) {
  const [contentIndex, setContentIndex] = useState(0)
  const { setCarouselCurrentIndex, classes } = props
  return (
    <Carousel
      navButtonsAlwaysVisible
      index={contentIndex}
      onChange={(index) => {
        setCarouselCurrentIndex(index)
      }}
      activeIndicatorProps={{
        className: classes.activeIndicator,
      }}
      indicatorProps={{
        className: classes.inactiveIndicator,
      }}
    >
      <BusinessCarouselFirstContent {...props} setContentIndex={setContentIndex} />
      <BusinessCarouselSecondContent {...props} setContentIndex={setContentIndex} />
      <BusinessCarouselThirdContent {...props} setContentIndex={setContentIndex} />
    </Carousel>
  )
}

BusinessPlanCarousel.propTypes = {
  classes: PropTypes.object,
  setCarouselCurrentIndex: PropTypes.func,
}

export default withWidth()(BusinessPlanCarousel)
