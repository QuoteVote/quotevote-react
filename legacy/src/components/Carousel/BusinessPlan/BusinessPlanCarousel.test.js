import React, { useState } from 'react'
import { render } from '@testing-library/react'

import { makeStyles, MuiThemeProvider as ThemeProvider } from '@material-ui/core/styles'
import BusinessPlanCarousel from './BusinessPlanCarousel'
import theme from '../../../themes/MainTheme'
import styles from '../../../assets/jss/material-dashboard-pro-react/views/landingPageStyle'

const useStyles = makeStyles(styles)

function BusinessPlanCarouselWrapper() {
  const classes = useStyles()
  const [carouselCurrentIndex, setCarouselCurrentIndex] = useState(0)
  return (

    <ThemeProvider theme={theme}>
      <BusinessPlanCarousel classes={classes} setCarouselCurrentIndex={setCarouselCurrentIndex} carouselCurrentIndex={carouselCurrentIndex} />
    </ThemeProvider>
  )
}

describe('BusinessPlanCarousel test -', () => {
  it('renders correctly', () => {
    const { container } = render(<BusinessPlanCarouselWrapper />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
