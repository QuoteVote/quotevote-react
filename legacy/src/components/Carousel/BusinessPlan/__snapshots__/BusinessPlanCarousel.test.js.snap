// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessPlanCarousel test - renders correctly 1`] = `null`;
